core/src/main/java/com/es/core/model/phone/JdbcPhoneDao.java

```diff
@@ -9,19 +8,72 @@
 import java.util.Optional;
 
 @Component
-public class JdbcPhoneDao implements PhoneDao{
+public class JdbcPhoneDao implements PhoneDao {
     @Resource
     private JdbcTemplate jdbcTemplate;
 
+    private static final String PHONES_AND_COLORS_QUERY = "SELECT phones.id, phones.brand, phones.model, phones.price, " +
+            "phones.imageUrl, colors.code, phone2color.colorId " +
+            "FROM phones " +
+            "JOIN phone2color ON phones.id=phone2color.phoneId " +
+            "JOIN colors ON phone2color.colorId=colors.id";
+    private static final String PHONES_AND_COLORS_QUERY_WITH_LIMIT_AND_OFFSET = PHONES_AND_COLORS_QUERY + " offset ? limit ?";
+    private static final String INSERT_INTO_PHONES2COLORS = "INSERT INTO phone2color VALUES (?, ?)";
+    private static final String DELETE_FROM_PHONES2COLORS = "DELETE FROM phone2color WHERE colorId = ? and phoneId = ?";
+    private static final String UPDATE_PHONE_PRICE = "UPDATE phones SET price = ?";
+    private static final String INSERT_NEW_PHONE = "INSERT INTO phones (brand, model, price, imageUrl) values (?, ?, ?, ?)";
+
     public Optional<Phone> get(final Long key) {
-        throw new UnsupportedOperationException("TODO");
+        return getPhoneListWithColors().stream()
+                .filter(phone -> phone.getId().equals(key))
+                .findAny();
+    }
+
+    private List<Phone> getPhoneListWithColors() {
+        return jdbcTemplate.query(PHONES_AND_COLORS_QUERY, new PhoneExtractor());
     }
 
     public void save(final Phone phone) {
-        throw new UnsupportedOperationException("TODO");
+        List<Phone> phoneList = getPhoneListWithColors();
```

Comment:
Saving wasn't the part of the task. Overall advice: try not to spend much time on the functionality that is not required at the moment :)
Requirements may change, approach may change, and you'll end up spending time on the code that will need to be removed.

Reply:
And also some comments to the way you implemented saving. Though you might need to remove it at all, or come back to it later, so it's more for you to avoid mistakes in future :)
First, you do a lot of unnecessary DB calls to insert the phone, and my previous comment about fetching all phones to get/update one is valid here as well.
Second, what if one of the inserts that you do here will fail? You'll end up with partial phone data in your DB, so it won't be consistent. Inserting/updating of entity usually need to be atomic operation - it should happen fully or it shouldn't happen at all throwing error. Transactions used in this case for that purpose